tests/functional_tests/ffront_tests/test_foast_parsing.py

```diff
@@ -266,6 +275,54 @@ def conditional_wrong_arg_type(
     assert re.search(msg, exc_info.value.__cause__.args[0]) is not None
 
 
+def test_ternary_with_field_condition():
+    def ternary_with_field_condition(cond: Field[[], bool]):
+        return 1 if cond else 2
+
+    with pytest.raises(FieldOperatorTypeDeductionError, match=r"should be .* `bool`"):
+        _ = FieldOperatorParser.apply_to_function(ternary_with_field_condition)
+
+
+def test_correct_return_type_annotation():
+    """See ADR 11."""
+
+    def correct_return_type_annotation() -> float:
+        return 1.0
+
+    FieldOperatorParser.apply_to_function(correct_return_type_annotation)
+
+
+def test_wrong_return_type_annotation():
+    """See ADR 11."""
+
+    def wrong_return_type_annotation() -> Field[[], float]:
+        return 1.0
+
+    with pytest.raises(FieldOperatorTypeDeductionError, match="Expected `bool`"):
+        _ = FieldOperatorParser.apply_to_function(wrong_return_type_annotation)
```

Comment:
Does this work? isn't that a strange error message? why bool? shouldn't it rather be something 'like Expected float'?

I had to set something like 

`    with pytest.raises(GTTypeError, match="Expected `float64`"):
        _ = FieldOperatorParser.apply_to_function(wrong_return_type_annotation)`

Reply:
Thanks, good catch. There is another test later in the file with the same name, therefore this is never executed. I fixed the test.